dotcom-rendering/src/lib/memoize.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Cache the results of calling a function with string keys
+ */
+export const memoize = <T>(f: (key: string) => T): ((key: string) => T) => {
+	const cache: {
+		[key: string]: T;
+	} = {};
```

Comment:
Completely up to you but icydk [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) tends to have a better performance profile than using objects as a dictionary. This use case might not benefit much though as it is read heavy.

https://www.builder.io/blog/maps
https://www.zhenghao.io/posts/object-vs-map#tldr

Reply:
Thanks for the links! I was sure I'd read somewhere that objects tend to perform better than maps for small datasets, but I can't find the source! I think maps are more semantic here anyway, so I'll switch it over.